cmd/scollector/collectors/haproxy_unix.go

```diff
@@ -0,0 +1,518 @@
+package collectors
+
+import (
+	"encoding/csv"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+var trackedInstances = []*trackedInstance{}
+
+type trackedInstance struct {
+	Tier     string
+	URL      string
+	Username string
+	Password string
+}
+
+// HAProxy registers an HAProxy collector.
+func HAProxy(user, pwd string) {
+	collectors = append(collectors, &IntervalCollector{
+		F: func() (opentsdb.MultiDataPoint, error) {
+			return cHAProxyCSV(user, pwd)
+		},
+		name: "haproxy",
+	})
+}
+
+// AddHAProxyInstance adds haproxy instances to be tracked.
+func AddHAProxyInstance(conf string) error {
+	sp := strings.SplitN(conf, ":", 2)
+	if len(sp) < 2 {
+		return fmt.Errorf("haproxy_instance requires two fields")
+	}
+	var instance trackedInstance
+	instance.Tier = sp[0]
+	instance.URL = sp[1]
+	trackedInstances = append(trackedInstances, &instance)
+
+	return nil
+}
+
+func cHAProxyCSV(user, pwd string) (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+	var err error
+	client := http.Client{}
```

Comment:
i don't see a reason the http client can't be created inside of `fetchHAProxyData`